README.md

```diff
@@ -9,92 +9,388 @@
-const togglerStore = require('../stores/togglerStore')
+Preferably we don't always want to implement our own eventing system every time
+we write a custom store, so in this case, we're simply going to inherit from
+`EventEmitter` (available via `events`) and use a single `change` event:
 
-module.exports = class Toggler extends React.Component {
-  constructor(props) {
-    super(props)
+```js
+const EventEmitter = require('events').EventEmitter;
 
-    this.handleClick = togglerActions.handleClick.bind(this)
-    this.updateState = this.updateState.bind(this)
+class BoardStore extends EventEmitter {
+  constructor (initialState = { columns: [] }) {
+    this.state = initialState;
+    super();
+  }
 
-    this.state = togglerStore.state
+  addListener (listener) {
+    EventEmitter.prototype.addListener.call(this, 'change', listener)
   }
 
-  componentDidMount() {
-    togglerStore.addListener(this.updateState)
+  removeListener (listener) {
+    EventEmitter.prototype.removeListener.call(this, 'change', listener)
   }
 
-  render() {
-    return <a onClick={this.handleClick}>{this.state.on}</a>
+  setState (state) {
+    this.emit('change', state);
+    this.state = state;
   }
 
-  updateState(state) {
-    // we can also perform other transformations here
-    this.setState(state)
+  getState () {
+    return this.state;
   }
 }
 ```
 
-This is an admittedly contrived example, but it lets us bypass the dispatcher
-(for now) and demonstrates the data flow in what is hopefully a reasonably
-obvious way.
+**Advanced** Having a single store might not always be the most desirable
+solution. "Classical" Flux can be implemented using multiple stores.
+
+And... BOOM! We have our store! Now let's have a look at our `<App />` component
+and wire it up!
+
+### Subscribing to store changes `<App />`
+
+Our `<App />` component is simply going to listen for store changes and
+encapsulate the corresponding application state whenever a store change occurs:
+
+```js
+class App extends React.Component {
+  constructor (props) {
+    // ...
+    this.listener = this.listener.bind(this);
+    this.setState({ board: boardStore.getState() });
+  }
+  componentDidMount () {
+    boardStore.addListener(this.listener);
+  }
+  componentWillUnmount () {
+    // We shouldn't forget this! Otherwise we would get a memory leak!
+    boardStore.removeListener(this.listener);
+  }
+  listener (board) {
+    // Update `<App />` state and trigger a re-render.
+    this.setState({ board });
+  }
+  // ...
+}
+```
+
+Our card can now update the store whenever someone changes its title. For now,
+we're going to add a method on the `BoardStore` to handle this logic. In
+subsequent lessons we're going to extract this update logic out event further.
+
+```js
+class BoardStore extends EventEmitter {
+  // ...
+  updateCardTitle (cardId, updatedTitle) {
+    for (const column of this.state.columns) {
+      for (const card of column.cards) {
```

Comment:
Rather than looping the cards, I think we can use `.find()` here:

```js
const cardtoModify = column.cards.find(card => card.id === cardId);
cardToModify.title = updatedTitle;
```